src/design/architectural-design.tex

```diff
@@ -66,3 +66,41 @@ \subsection{Criterion}\label{subsec:criterion}
 % textidote: ignore begin
 \subsection{Architecture}\label{subsec:architecture}
 % textidote: ignore end
+
+\begin{figure}[H]
+    \centering
+    \includegraphics[width=.5\textwidth]{architecture-overview.png}
+    \caption{An overview of the system architecture.
+    }\label{fig:architecture-overview}
+\end{figure}
+
+\begin{figure}[H]
+    \centering
+    \includegraphics[width=\textwidth]{architecture-backend.png}
+    \caption{The back end architecture.
+    }\label{fig:architecture-backend}
+\end{figure}
```

Comment:
Can we use "h"'s here instead of "H"'s?

Reply:
Using "h" breaks the layout. I think "H" looks better. But image placement is something we can consider at the end of the report.